### PR TITLE
Use this.emitFile to generate css files

### LINKED ISSUE #71: Use this.emitFile to output css
https://github.com/rollup/rollup-plugin-svelte/blob/f8e8fa15e5cdf87847a2b02a5471f9f4481f8edb/index.js#L297

Saw this comment, I have been using this and I think it's relevant to that comment.

```js
generateBundle() {
  this.emitFile({type: 'asset', fileName: 'some-file.scss', source: sourceCode});
},
```

More info here, https://rollupjs.org/guide/en/#thisemitfileemittedfile-emittedchunk--emittedasset--string hope this helps.

### PR DESCRIPTION
**Pull request by shirotech**
Fixes #71

### CODE REVIEW

**Comment by elgertam** [comment]
This PR would be quite useful for a project I'm working on. I'm using Gulp with Rollup, and using `fs.writeFileSync()` in `CssWriter.write()` is a bit unexpected. Would strongly prefer to follow the Rollup conventions than to modify the filesystem in place.

**Comment by shirotech** [comment]
@elgertam I believe rollup can fully replace gulp, is there a specific reason you're using them together?

**Comment by elgertam** [comment]
TL;DR: Gulp is very simple yet powerful, and is also debuggable. Rollup has a bit more complexity, but does code bundling very well.

I find that Gulp makes the build process very simple and understandable both to my team and me. I can commit a `gulpfile.js`, and my team of engineers and DevOps can understand it. If any of us doesn't understand something gulp-related, a 5-minute review of the documentation can typically resolve the issue. If that doesn't help, we can add a breakpoint in our `gulpfile.js` and begin debugging at almost any arbitrary point without much effort. Further, if there is some capability that doesn't exist as a Gulp plugin on NPM, I can usually roll something relatively quickly, given that Gulp is friendly with streams & Promises/async/await and has a simple and easy-to-understand interface. (Indeed, this is Rollup's recommended way to integrate with Gulp, and is how I accomplished it - no plugin required.)

Rollup is superb at orchestrating the build pipeline. But since it favors configuration slightly more than code (though not to the degree of, say, Webpack, which invokes magic to the point of sorcery), I find Rollup with a bunch of plugins a bit harder to navigate. While I am capable of digging through documentation and code for arbitrary plugins in order to figure out why my build isn't exactly what I expect, this process has a bit more friction in Rollup as the number of plugins increase. Also, if I have some task or process I wish to use that has not been implemented as a Rollup plugin, it's a bit harder for me to roll something myself, since the interface Rollup provides is quite a bit more complex than that for Gulp. (The complexity of Rollup's interface seems to be this PR's _raison d'être_.)

I strongly prefer my build code (and my production code) to be as simple as possible, but not simpler. I find Gulp + Rollup are a good mix.

**Comment by elgertam** [comment]
@shirotech I should add that regardless of my preferences, this PR fixes a fairly significant defect in the plugin. The behavior would be a bit unexpected to me even if I was using Rollup without Gulp. I hope it's merged soon.

**Comment by shirotech** [comment]
@elgertam sorry I was just curious, I see that makes sense now, I do admit the setup for rollup can be complex, I've spent countless hours just to make it simple to the point where I gave up relying on 3rd party plugins and rolling our own versions.

Back on topic, yes hope this can be merged soon.

**Comment by benmccann** [comment]
@shirotech I'd love to get this merged. Can you take a look at why the tests are failing?

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "eslint": "^6.8.0",
     "locate-character": "^2.0.5",
     "mocha": "^7.1.1",
-    "rollup": "^0.67.4",
+    "rollup": "^1.32.1",
```
Comment (benmccann) [inline package.json:32]: If we're bumping the Rollup version, should we bump it to 2.x?

Comment (shirotech) [reply package.json:32]: Well it was shbmitted 8 months ago and that was the latest version at the time, I can spend some time to upgrade to latest.

File: index.js [hunk 2]
```diff
@@ -327,11 +313,13 @@ module.exports = function svelte(options = {}) {
 					}
 				}
 
-				const writer = new CssWriter(result, {
+				const filename = Object.keys(bundle)[0].split('.').shift() + '.css';
```
Comment (benmccann) [inline index.js:316]: Should this do it for each element of the array? If not, it might be useful to add a comment explaining why just `[0]`

File: package.json (hunk 1 above)
Comment (benmccann) [reply package.json:32]: I tested and there's no code change required to bump it to 2.x. But maybe we should just leave it at 1.x since some users may not have upgraded yet

**Comment by benmccann** [comment]
@shirotech the fix to the test is relatively easy. You need to change the three calls to `css.write` in `test/test.js` to remove the path and just use the filename as you did in the README. E.g. change `css.write('test/filename-test/dist/bundle.css');` to `css.write('bundle.css');`

**Comment by benmccann** [comment]
@shirotech just checking in on this. do you think you'll be able to update the test with the fix I suggested?

**Comment by pngwn** [comment]
I fixed those tests, this looks good now.

File: package.json [hunk 3]
```diff
@@ -41,6 +41,6 @@
   },
   "peerDependencies": {
     "svelte": "*",
-    "rollup": ">=0.60.0"
+    "rollup": ">=1.32.1"
```
Comment (benmccann) [inline package.json:44]: this should probably be `>=1.0.0`

**Comment by Conduitry** [comment]
Is there a way to do this that's also compatible with older versions of Rollup?

If not, since this is a breaking change anyway, is this now using the latest version of Rollup's asset handling APIs, and not one of the interim ones?

**Comment by pngwn** [comment]
Apparently this repo is using a version below 1? I must be missing something, this doesn't seem possible.

**Comment by pngwn** [comment]
Or did it get updated in this PR?

**Comment by pngwn** [comment]
It did. Ignore me.

**Comment by pngwn** [comment]
I believe `emitFile` with type set to `’asset'` is the latest incarnation. `emitAsset` Was the previous one.

**Comment by benmccann** [comment]
I just tested. This PR only works with Rollup `>= 1.19.2`, which was released on Aug 5, 2019

**Comment by benmccann** [comment]
Merging this per discussion with @pngwn and @lukeed

**Comment by mustafa0x** [comment]
The change in this PR is breaking to 100% of users. Something needs to be done to notify them. For example, if there is a directory separator or if the directory to write to doesn't exist, warn that paths are now relative.
